components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/GatewayArtifactsMgtDAO.java

```diff
@@ -643,6 +643,7 @@ public void removeOrganizationGatewayArtifacts(String organization) throws APIMa
                     SQLConstants.DELETE_BULK_GW_PUBLISHED_API_DETAILS)) {
                 preparedStatement.setString(1, organization);
                 preparedStatement.executeUpdate();
+                artifactSynchronizerConn.setAutoCommit(false);
```

Comment:
why we add setAutoCommit(false); shouldn't it be added just after taking the connection?